example/gc_metrics.ml

```diff
@@ -0,0 +1,32 @@
+open Lwt.Infix
+
+let allocate ~wait n () =
+  (* allocate an array of size [n] *)
+  Lwt_io.printf "allocating an array of size %d\n" n >>= fun () ->
```

Comment:
curious about this kind of pattern. haskell has addressed this with a `do` notation and I wan under impression that's how all these `let|` , `let@` etc variations were used.
why not do binding with
`let| _ = Lwt_io.printf "allocating an array of size %d\n" n` 
or somesuch? Or is this just a preferred style?

Reply:
This is the same idea, it just uses infix operators rather than the new let bindings. This library was started before the `let` bindings were supported, so it includes a lot of the infix operators.